src/bootrom.rs

```diff
@@ -0,0 +1,73 @@
+use core::mem;
+
+// A bootrom function table code.
+pub type RomFnTableCode = [u8; 4];
+
+// Defines a code.
+macro_rules! table_codes {
+    ($($name:ident, $code:expr;) +) => {
+        $(pub const $name: RomFnTableCode = $code;) +
+    };
+}
+
+// TODO: wait for newer rust version where stringify works on bytes.
+table_codes! {
+    // 2.8.3.1.3. Flash Access Functions
+    IF, *b"\0\0IF";
+    EX, *b"\0\0EX";
+    RE, *b"\0\0RE";
+    RP, *b"\0\0RP";
+    FC, *b"\0\0FC";
+    CX, *b"\0\0CX";
+}
+
+const ROM_TABLE_LOOKUP_PTR: *const u16 = 0x18 as _;
+const OPS_TABLE_PTR: *const u16 = 0x14 as _;
+
+/// This function searches for (table)
+type RomTableLookupFn<T> = unsafe extern "C" fn(*const u16, u32) -> T;
+
+/// Given a rom_address pointer, convert a 16 bit pointer stored
+/// at the given rom address into a 32 bit pointer
+unsafe fn get_ptr_from_rom(rom_address: *const u16) -> *const u16 {
```

Comment:
This function seems to be redundant

Reply:
It's not, it is explicit. This function should be inlined.

Reply:
:+1: get it, missed the dereference